izer/quantize.py

```diff
@@ -80,9 +80,10 @@ def mean_n_stds_max_abs(t, n_stds=1):
     def get_const(_):
         return arguments.scale
 
-    def get_max_bit_shift(t, shift_quantile, return_bit_shift=False):
+    def get_max_bit_shift(t, clamp_bits, shift_quantile, return_bit_shift=False):
         float_scale = 1.0 / torch.quantile(t.abs(), shift_quantile)
-        bit_shift = torch.floor(torch.log2(float_scale))
+        bit_shift = torch.floor(torch.log2(float_scale)) \
+            .clamp(min=-7.-clamp_bits, max=23.-clamp_bits)
```

Comment:
The original mechanism in QAT also has hard-coded limits in the calculation (-15 and 15 currently for QAT, but I'll create a pull request there as well, to make it adaptable for different quantization bits).

Perhaps these hard-coded numbers could be defined via constant variables at the beginning of the file. Opinions?